JetNews/app/src/main/java/com/example/jetnews/ui/JetnewsApp.kt

```diff
@@ -53,22 +58,40 @@ fun JetnewsApp(
 
             val navBackStackEntry by navController.currentBackStackEntryAsState()
             val currentRoute = navBackStackEntry?.destination?.route ?: MainDestinations.HOME_ROUTE
-            Scaffold(
-                scaffoldState = scaffoldState,
-                drawerContent = {
-                    AppDrawer(
-                        currentRoute = currentRoute,
-                        navigateToHome = { navController.navigate(MainDestinations.HOME_ROUTE) },
-                        navigateToInterests = { navController.navigate(MainDestinations.INTERESTS_ROUTE) },
-                        closeDrawer = { coroutineScope.launch { scaffoldState.drawerState.close() } }
+
+            BoxWithConstraints {
+                val windowSize = getWindowSize(maxWidth)
+                Scaffold(
+                    scaffoldState = scaffoldState,
+                    // If the window size is Compact, show the AppDrawer.
+                    // Otherwise, a NavRail will be shown in individual screens
+                    drawerContent = if (windowSize == WindowSize.Compact) {
+                        {
+                            AppDrawer(
+                                currentRoute = currentRoute,
+                                navigateToHome = {
+                                    navController.navigate(MainDestinations.HOME_ROUTE)
+                                },
+                                navigateToInterests = {
+                                    navController.navigate(MainDestinations.INTERESTS_ROUTE)
+                                },
+                                closeDrawer = {
+                                    coroutineScope.launch { scaffoldState.drawerState.close() }
+                                }
+                            )
+                        }
+                    } else {
+                        null
+                    }
+                ) { innerPaddingModifier ->
+                    JetnewsNavGraph(
+                        appContainer = appContainer,
+                        showNavRail = windowSize != WindowSize.Compact,
```

Comment:
This is just an approach thing, but what made you decide to pass down the resolved `showNavRail` here, versus the `windowSize`?

Reply:
+1 to keeping things primitive

Reply:
`showNavRail` is a better API in this case as it's clear what's the intend behind it vs `windowSize` which is not clear what the composable should behave. It's also easier for testing and previewing this composable.

Reply:
What do you mean with "primitive" @JoseAlcerreca?

Reply:
I might be reading ahead to the list/detail case, but keeping it as the `windowSize` allows any route to decide individually what to do based on the `windowSize`, instead of being told what it has to do. I don't feel super strongly though either way.

Reply:
Let's leave it like this for this PR, and if there are reasons for change it in the future, let's do so. I prefer to keep it like this for this particular PR because of the reasons mentioned above

Reply:
I mean I prefer to pass a well-named boolean instead of a windowSize, whose reason to be there is not clear.

Reply:
Yeah! Agreed, so no changes required here